tests/ui/strlen_on_c_strings.stderr

```diff
@@ -0,0 +1,16 @@
+error: using `libc::strlen` on a `CString` or `CStr` value
+  --> $DIR/strlen_on_c_strings.rs:11:24
+   |
+LL |     let len = unsafe { libc::strlen(cstring.as_ptr()) };
+   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `cstring.as_bytes().len()`
```

Comment:
The suggested code don't need unsafe block. Can unsafe block be checked? Or this comments needs to be improved to include removing unsafe block.

Reply:
Good catch, yes, unsafe block is redundant here. It can be removed when it's the only one unsafe operation inside of it.
What about for now to lint like this: "help: try this: `cstring.as_bytes().len()`, you might also need to get rid of `unsafe` block in some cases"?

Reply:
Yes, it looks good as a first step! That unsafe block might be removed is one of purposes of this lint.

If unsafe block has only `libc::strlen(cstr.as_ptr())`, it would be better to remove unsafe block in suggestion. But I think it can be fixed with another follow-up PR.

Reply:
Created the issue for that: https://github.com/rust-lang/rust-clippy/issues/7436